pkg/agent/agent.go

```diff
@@ -466,8 +483,14 @@ func (i *Initializer) setupDefaultTunnelInterface(tunnelPortName string) error {
 		}
 	}
 
-	// Create default tunnel port.
+	// Create the default tunnel port and interface.
 	if i.networkConfig.TrafficEncapMode.SupportsEncap() {
+		if tunnelPortName != defaultTunInterfaceName {
+			// Reset the tunnel interface name to the configured name before
```

Comment:
to the **desired** name to avoid confusion?

Reply:
Sure.